connector/spanmetricsconnectorv2/histogram.go

```diff
@@ -0,0 +1,155 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package spanmetricsconnectorv2 // import "github.com/elastic/opentelemetry-collector-components/connector/spanmetricsconnectorv2"
+
+import (
+	"context"
+	"errors"
+	"sort"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatautil"
+)
+
+var noAttributes = [16]byte{}
+
+func newExplicitHistogram(metricDefs map[string]metricDef) *explicitHistogram {
+	return &explicitHistogram{
+		metricDefs: metricDefs,
+		counts:     make(map[string]map[[16]byte]*attrExplicitHistogram, len(metricDefs)),
+		timestamp:  time.Now(),
+	}
+}
+
+type explicitHistogram struct {
+	metricDefs map[string]metricDef
+	counts     map[string]map[[16]byte]*attrExplicitHistogram
+	timestamp  time.Time
+}
+
+type attrExplicitHistogram struct {
+	attrs pcommon.Map
+
+	sum   float64
+	count uint64
+
+	bounds []float64
+	counts []uint64
+}
+
+func newAttrExplicitHistogram(attrs pcommon.Map, bounds []float64) *attrExplicitHistogram {
+	return &attrExplicitHistogram{
+		attrs:  attrs,
+		bounds: bounds,
+		counts: make([]uint64, len(bounds)+1),
+	}
+}
+
+func (c *explicitHistogram) update(ctx context.Context, attrs pcommon.Map, value float64) error {
+	var multiError error
+	for name, md := range c.metricDefs {
+		countAttrs := pcommon.NewMap()
+		for _, attr := range md.attrs {
```

Comment:
I suppose it's OK for now, and we can refactor when adding summary & exponential histogram support, but this seems like it should be up a level in the main connector code, rather than in the explicitHistogram code.

Reply:
Yes, I plan to abstract explicit histogram, exponential histogram, and summary behind a common interface.